packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -453,6 +460,29 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
           columns: visibleColumns,
         });
       }
+
+      // If the selected row is not within the current range of rows being displayed,
+      // we need to render it at either the top or bottom of the rows,
+      // depending on whether it is above or below the range.
+      if (!isFocusedCellRendered && cell && !disableVirtualization) {
+        const rows = currentPage.rows.filter((row) => row.id === cell.id);
+        const focusedRow = rows.length > 0 && rows[0];
+
+        if (focusedRow) {
+          const index = currentPage.rows.indexOf(focusedRow);
+          if (index > firstRowToRender) {
+            renderedRows.push(focusedRow);
+          } else {
+            renderedRows.unshift(focusedRow);
+          }
+          apiRef.current.calculateColSpan({
```

Comment:
It doesn't seem to be working with pinned columns (and column spanning?):
https://deploy-preview-7357--material-ui-x.netlify.app/x/react-data-grid/column-spanning/#function-signature


https://user-images.githubusercontent.com/13808724/212136172-443d0379-f8dd-4be6-b9d7-3ceb61834322.mov